pkg/status/handlers.go

```diff
@@ -17,7 +18,37 @@ type Status struct {
 	BuildInfo *BuildInfo `json:"buildInfo"`
 }
 
+type DeepCheckStatus struct {
+	KratosStatus string `json:"kratos_status"`
+	HydraStatus  string `json:"hydra_status"`
+}
+
+type SafeBool struct {
+	mu    sync.Mutex
+	value bool
+}
+
+func (s *SafeBool) Set(value bool) {
+	s.mu.Lock()
+	s.value = value
+	s.mu.Unlock()
+}
+
+func (s *SafeBool) Get() bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.value
+}
+
+func newSafeBool(starter bool) *SafeBool {
+	return &SafeBool{
+		value: starter,
+	}
+}
```

Comment:
do we need this?

Reply:
some earlier push could have had race condition, but right now I don't think that's plausible, because we only write to the bools one in the go routines, and only read from them after the goroutines are done. I took out this struct.